plugins/processors/dcos_metadata/dcos_metadata.go

```diff
@@ -296,16 +299,26 @@ func mapExecutorNames(ge *agent.Response_GetExecutors) map[string]string {
 	return results
 }
 
+func contains(list []string, target string) bool {
+	for _, str := range list {
+		if str == target {
+			return true
+		}
+	}
+	return false
+}
+
 // mapTaskLabels returns a map of all task labels prefixed DCOS_METRICS_
-func mapTaskLabels(labels *mesos.Labels) map[string]string {
+// or included in list of whitelisted labels
+func mapTaskLabels(labels *mesos.Labels, whitelist []string) map[string]string {
 	results := map[string]string{}
 	if labels != nil {
 		for _, l := range labels.GetLabels() {
 			k := l.GetKey()
-			if len(k) > 13 {
-				if k[:13] == "DCOS_METRICS_" {
-					results[k[13:]] = l.GetValue()
-				}
+			if len(k) > 13 && k[:13] == "DCOS_METRICS_" {
```

Comment:
Would it make sense to use [`strings.HasPrefix`][(https://golang.org/pkg/strings/#HasPrefix) here instead of checking the elements directly? This would automatically handle the incorrect length too:
```suggestion
			if strings.HasPrefix(k, "DCOS_METRICS_") {
```

Reply:
Or is there some performance implication?

Reply:
Looks like it has the [same implementation](https://golang.org/src/strings/strings.go?s=11269:11306#L440) :) will change!